### PR TITLE
IWebHostEnvironment for Startup ctor injection 3.0

### LINKED ISSUE #15257: Unintentional reference to deprecated IHostingEnvironment
Hello! The following documentation note about HostBuilder says it injects IHostingEnvironment into Startup's constructor, which is deprecated. I'm guessing you meant to say IWebHostEnvironment instead?

> The most significant change from WebHostBuilder to HostBuilder is in dependency injection (DI). When using HostBuilder, you can only inject IConfiguration and IHostingEnvironment into Startup's constructor. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 76f39eec-9dd1-6167-56ec-47b26e0b5063
* Version Independent ID: f9f2b0f3-01eb-cf58-8767-194fe0fc0e44
* Content: [Migrate from ASP.NET Core 2.2 to 3.0](https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.0&tabs=visual-studio#migrate-startupconfigure)
* Content Source: [aspnetcore/migration/22-to-30.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/migration/22-to-30.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-migration**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #15257

https://github.com/aspnet/Announcements/issues/353

Thanks @IGx89! 🎸

### CODE REVIEW

File: aspnetcore/migration/22-to-30.md [hunk 1]
```diff
@@ -1025,7 +1025,13 @@ The following code shows the ASP.NET Core 2.2 template-generated `Program` class
 
 <xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder> remains in 3.0 and is the type of the `webBuilder` seen in the preceding code sample. <xref:Microsoft.AspNetCore.Hosting.WebHostBuilder> will be deprecated in a future release and replaced by `HostBuilder`.
 
-The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). When using `HostBuilder`, you can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration> and <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment> into `Startup`'s constructor. The `HostBuilder` DI constraints:
+The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). When using `HostBuilder`, you can only inject the following into `Startup`'s constructor:
+
+* <xref:Microsoft.Extensions.Configuration.IConfiguration>
+* <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment>
```
Comment (serpent5) [inline aspnetcore/migration/22-to-30.md]: The announcement has this one as `IHostEnvironment`. `IHostingEnvironment` is obsolete as of 3.0.

Comment (guardrex) [reply aspnetcore/migration/22-to-30.md]: ooops! Good :eye: @serpent5!

Comment (guardrex) [reply aspnetcore/migration/22-to-30.md]: I'm going to 1/2 take the hit on that one .... **_and 1/2 blame Azure AD!!!!!_** 😠 :smile:

**Comment by guardrex** [comment]
DAH ... and without 3.0 API, it hates that link. 🤦‍♂

**Comment by guardrex** [comment]
Wait ... is that in Extensions. I think so!

**Comment by serpent5** [comment]
Yeah, it is. Was just about to say...

File: aspnetcore/migration/22-to-30.md [hunk 2]
```diff
@@ -1025,7 +1025,13 @@ The following code shows the ASP.NET Core 2.2 template-generated `Program` class
 
 <xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder> remains in 3.0 and is the type of the `webBuilder` seen in the preceding code sample. <xref:Microsoft.AspNetCore.Hosting.WebHostBuilder> will be deprecated in a future release and replaced by `HostBuilder`.
 
-The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). When using `HostBuilder`, you can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration> and <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment> into `Startup`'s constructor. The `HostBuilder` DI constraints:
+The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). When using `HostBuilder`, you can only inject the following into `Startup`'s constructor:
+
+* <xref:Microsoft.Extensions.Configuration.IConfiguration>
+* `Microsoft.Extensions.Hosting.IHostEnvironment`
```
Comment (serpent5) [inline aspnetcore/migration/22-to-30.md]:
```suggestion
* <xref:Microsoft.Extensions.Hosting.IHostEnvironment>
```

Comment (serpent5) [reply aspnetcore/migration/22-to-30.md]: The xref seems to work when used with the `Extensions` namespace. I checked this on my machine using docfx, but I dunno if there's more to it than that.

Comment (guardrex) [reply aspnetcore/migration/22-to-30.md]: Can't do that yet ... no 3.0 API yet ... they're working on it.

Comment (serpent5) [reply aspnetcore/migration/22-to-30.md]: Just for completeness, this is where it takes me: https://docs.microsoft.com/en-gb/dotnet/api/microsoft.extensions.hosting.ihostenvironment?view=dotnet-plat-ext-3.0.

Comment (guardrex) [reply aspnetcore/migration/22-to-30.md]: Try this tho ... https://docs.microsoft.com/en-us/dotnet/api/?term=Microsoft.Extensions.Hosting.IHostEnvironment

Comment (guardrex) [reply aspnetcore/migration/22-to-30.md]: My understanding as of yesterday is that there's more work to be done in docs engineering on it. We have an open issue to update API links when we know that they've given the word that it's ready. Until then, I think the code fence is ok.

Comment (serpent5) [reply aspnetcore/migration/22-to-30.md]: Yeah, I understand. I provided the link just so you could see what I saw, but I never doubted that you knew better than I did. I just wondered if you'd missed the xref after changing the namespace. Sorry to have distracted you from your Azure AD party. :smiley:

Comment (guardrex) [reply aspnetcore/migration/22-to-30.md]: Azure AD Party 😄😢😄😠😄 ..... they must know that Halloween is here and that I needed a good *SCARE* today. 👻🎃

I have half of it working ... Managed Identities :+1: ... but Azure AD with a cert :-1:. ei ei ei ... I'll figure it out ... beat it into submission.

.... hummmmmm 🤔 ..........

I wonder if there are any open positions over at **_Apple docs_**??!!! :smile:

**Pull request merged**